src/hipscat/catalog/association_catalog/partition_join_info.py

```diff
@@ -30,8 +39,82 @@ def _check_column_names(self):
             if column not in self.data_frame.columns:
                 raise ValueError(f"join_info_df does not contain column {column}")
 
+    def primary_to_join_map(self) -> Dict[HealpixPixel, List[HealpixPixel]]:
+        """Generate a map from a single primary pixel to one or more pixels in the join catalog.
+
+        Lots of cute comprehension is happening here, so watch out!
+        We create tuple of (primary order/pixel) and [array of tuples of (join order/pixel)]
```

Comment:
Could we add the docstring args and return type?

Reply:
There are no args for this one - just uses data from `self`.